app/src/main/java/com/example/dndwiki/database/DBSource.kt

```diff
@@ -23,7 +27,15 @@ class DBSource(
         return spellDetailsDao.findByIndex(index).toSpellDetails()
     }
 
-    suspend fun saveSpellDetails(spellDetails: SpellDetails) {
-        spellDetailsDao.insert(spellDetails.toSpellDetailsDB())
+    suspend fun saveAllSpellDetails(spellDetails: List<SpellDetails>) {
+        val spellDetailsDB = mutableListOf<SpellDetailsDB>()
+        withContext(Dispatchers.IO) {
```

Comment:
No need for this... its just mapping 1 object to another

you can just call  spellDetails.map { .. }